### PR TITLE
Blog post structure up, first post to be up soon

### PR DESCRIPTION
**Pull request by aliahsan07**

### CODE REVIEW

File: content/blog-posts/2020-02-12-js-the-definitive-guide.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+I've started reading the book JavaScript: The Definitive Guide (5th edition) as part of my research work to better understand the intricate details of language features. I know the book is over 10 years old and JS has evolved so much since then. But my goal is more oriented towards understanding the core JavaScript features and how JQuery comes into the game exploiting many of those properties. I'll probably then build up on them with further editions of the book but I'm not really focused on the ES6 world as of now.
+
+I'm going to list some of the findings from the book:
+
+- JS interpreter can run scripts in different global contexts and those contexts need not be entirely separate; they can refer to each other. But this possesses a security problem.
+- About variable scoping: every variable is defined in a scope chain. For a nested function, multiple objects will be in a scope chain. Lets say x is defined inside a nested function inside a function. The scope chain starts from nested function to function to global context and when x scope has to be found it starts from the bottom (that is nested fn). It goes up all the way to global context, if not found there its undefined.
```
Comment (AbreezaSaleem) [inline content/blog-posts/2020-02-12-js-the-definitive-guide.md]: it'll be more clear if u gave these functions some random names. Like A and B.

Comment (aliahsan07) [reply content/blog-posts/2020-02-12-js-the-definitive-guide.md]: yup its wip. its failing all the tests currently.

Comment (AbreezaSaleem) [reply content/blog-posts/2020-02-12-js-the-definitive-guide.md]: why?

File: content/blog-posts/2020-02-12-js-the-definitive-guide.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+2. Structured and thoughtful blog posts: this is the normal kind of blog posts which is ubiquitous on the internet. Most of the stuff here will come from the world of software development especially React ecosystem which I so dearly loved when I was in the industry.
+
+Now back to the original topic.
+
+I've started reading the book JavaScript: The Definitive Guide (5th edition) as part of my research work to better understand the intricate details of language features. I know the book is over 10 years old and JS has evolved so much since then. But my goal is more oriented towards understanding the core JavaScript features and how JQuery comes into the game exploiting many of those properties. I'll probably then build up on them with further editions of the book but I'm not really focused on the ES6 world as of now.
+
+I'm going to list some of the findings from the book:
+
+- JS interpreter can run scripts in different global contexts and those contexts need not be entirely separate; they can refer to each other. But this possesses a security problem.
```
Comment (AbreezaSaleem) [inline content/blog-posts/2020-02-12-js-the-definitive-guide.md]: this look a bit abstract. perhaps give an example here?

**Pull request merged**